# Aritmética y variables

Este ejercicio le permitirá comenzar a ejecutar su propio código.
Configure el cuaderno

Para comenzar, ejecute el código en la siguiente celda.

Comience haciendo clic dentro de la celda del código.
Haga clic en el triángulo (con la forma de un "botón de reproducción") que aparece a la izquierda de la celda del código.
Si su código se ejecutó correctamente, verá Configuración completa como resultado debajo de la celda.

En lugar de hacer clic en el triángulo, también puede ejecutar el código presionando Shift + Enter en su teclado. ¡Pruébelo ahora! No ocurrirá nada malo si ejecuta el código más de una vez.

### Pregunta 1

A continuación, ejecutará parte del código del tutorial para que pueda ver cómo funciona. Ejecute la siguiente celda de código sin cambios.

In [ ]:
print("¡Hola mundo!")

Acabas de ejecutar el código para imprimir ¡Hola mundo!, que deberías ver en el resultado anterior.

### Pregunta 2

Ahora, imprimirá otro mensaje de su elección. Para ello, cambie print("¡Su mensaje aquí!") para utilizar un mensaje diferente. Por ejemplo, puede cambiarlo a algo como:

print("¡Buenos días!")
print("Estoy aprendiendo a codificar :D")

O puede ver qué sucede si escribe algo como print("3+4"). ¿Devuelve 7 o simplemente piensa en "3+4" como otro mensaje más?

Asegúrese de que su mensaje esté entre comillas (") y que el mensaje en sí no utilice comillas. Por ejemplo, esto generará un error: print("Ella dijo "gran trabajo" y me dio un "high-five!") porque el mensaje contiene comillas. Si decide realizar el curso de Python después de completar este curso, aprenderá más sobre cómo evitar este error en la Lección 6.

¡Siéntase libre de probar varios mensajes!

In [ ]:
# TODO: Change the message
print("¡Su mensaje aquí!")

### Pregunta 3

Como aprendiste en el tutorial, un comentario en Python tiene un signo de numeral (#) delante, que le indica a Python que ignore el texto que lo sigue.

Poner un signo de numeral delante de una línea de código hará que Python ignore ese código. Por ejemplo, Python ignoraría esta línea y no aparecería nada en el resultado:

* #print(1+2)

Quitar el signo de numeral hará que puedas ejecutar el código nuevamente. Cuando eliminamos el signo de numeral delante de una línea de código, lo llamamos descomentar.

En este problema, descomentarás dos líneas en la celda de código a continuación y verás el resultado:

Elimina el # de suma. Para evitar errores, NO elimines el # delante de # Descomentar para ver la sugerencia.
A continuación, elimina el # de print('El resultado de la suma es:', suma).


In [ ]:
# Descomentar para ver la sugerencia.

# suma = 3+4
print(suma)

# Desconeta para ver el resultado.

# print('El resultado de la suma es:', suma)

### Pregunta 4

En el tutorial, definió varias variables para calcular la cantidad total de segundos en un año. Ejecute la siguiente celda de código para realizar el cálculo aquí.

In [ ]:
# Create variables
num_years = 4
days_per_year = 365
hours_per_day = 24
mins_per_hour = 60
secs_per_min = 60

# Calculate number of seconds in four years
total_secs = secs_per_min * mins_per_hour * hours_per_day * days_per_year * num_years
print(total_secs)

Utilice la siguiente celda de código para:

Defina una variable births_per_min y configúrela en 250. (En promedio, nacen 250 bebés cada minuto).
Defina una variable births_per_day que contenga la cantidad promedio de bebés que nacen cada día. (Para establecer el valor de esta variable, debe utilizar births_per_min y algunas de las variables de la celda de código anterior).

In [ ]:
# TODO: Set the value of the births_per_min variable
births_per_min = ___

# TODO: Set the value of the births_per_day variable
births_per_day = ___


### 🌶️ Pregunta 5

(¡Las preguntas marcadas con un 🌶️ serán un poco más desafiantes que las demás! Recuerda que siempre puedes obtener una pista o ver la solución).

La competencia Titanic es la competencia de ciencia de datos más famosa de Kaggle. En esta competencia, los participantes tienen el desafío de construir un modelo de aprendizaje automático que pueda predecir si los pasajeros sobrevivieron o no al naufragio del Titanic, en función de información como la edad, el sexo, el tamaño de la familia y el número de boleto.

Ejecuta la siguiente celda de código sin cambios para cargar y obtener una vista previa de los datos del Titanic.

No te preocupes por los detalles del código por ahora: el resultado final es simplemente que todos los datos del Titanic se han cargado en una variable llamada titanic_data. (Para aprender a escribir este código tú mismo, puedes tomar el curso de Python y luego el curso de Pandas).


In [ ]:
pip install pandas

In [7]:
import pandas as pd
url = 'https://raw.githubusercontent.com/MicrosoftDocs/mslearn-introduction-to-machine-learning/main/Data/titanic.csv'
titanic_data = pd.read_csv(url)
titanic_data.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

Los datos tienen una fila diferente para cada pasajero.

La siguiente celda de código define e imprime los valores de tres variables:

* total = número total de pasajeros que abordaron el barco
* survivor = número de pasajeros que sobrevivieron al naufragio
* minors = número de pasajeros menores de 18 años

Ejecute la celda de código sin cambios. (No se preocupe por los detalles de cómo se calculan estas variables por ahora. Puede obtener más información sobre cómo calcular estos valores en el curso de Pandas).

In [5]:
# Number of total passengers
total = len(titanic_data)
print(total)

# Number of passengers who survived
survived = (titanic_data.Survived == 1).sum()
print(survived)

# Number of passengers under 18
minors = (titanic_data.Age < 18).sum()
print(minors)

891
342
113


Por lo tanto,

* total = 891 (había 891 pasajeros a bordo del Titanic),
* survived = 342 (342 pasajeros sobrevivieron) y
* minors = 113 (113 pasajeros eran menores de 18 años).

En la celda de código a continuación, reemplace los guiones bajos (____) con código para calcular los valores de dos variables más:

* survived_fraction debe configurarse como la fracción de pasajeros que sobrevivieron al desastre del Titanic.
* minors_fraction debe ser la fracción de pasajeros que eran menores (menores de 18 años).

Para cada variable, su respuesta debe ser un número entre 0 y 1.

In [ ]:
# TODO: Fill in the value of the survived_fraction variable
survived_fraction = survived/total
# Print the value of the variable
print(survived_fraction)

# TODO: Fill in the value of the minors_fraction variable
minors_fraction = minors/total

# Print the value of the variable
print(minors_fraction)

debes obtener:

* 0.3838383838383838
* 0.12682379349046016

De lo contrario, verifica tu codigo.